server/services/zigbee2mqtt/api/zigbee2mqtt.controller.js

```diff
@@ -0,0 +1,149 @@
+const asyncMiddleware = require('../../../api/middlewares/asyncMiddleware');
+const { EVENTS, WEBSOCKET_MESSAGE_TYPES } = require('../../../utils/constants');
+const logger = require('../../../utils/logger');
+
+module.exports = function Zigbee2mqttController(gladys, zigbee2mqttManager) {
+  /**
+   * @api {post} /api/v1/service/zigbee2mqtt/discover Launch Zigbee2mqtt devices discovery
+   * @apiName discover
+   * @apiGroup Zigbee2mqtt
+   */
+  async function discover(req, res) {
+    logger.log('Launching devices discovery');
+    await zigbee2mqttManager.discoverDevices();
+    res.json({ status: 'discovering' });
+  }
+
+  /**
+   * @api {get} /api/v1/service/zigbee2mqtt/status Get Zigbee2mqtt connection status
+   * @apiName status
+   * @apiGroup Zigbee2mqtt
+   */
+  async function status(req, res) {
+    logger.log('Get status');
+    const response = await zigbee2mqttManager.status();
+    res.json(response);
+  }
+
+  /**
+   * @api {post} /api/v1/service/zigbee2mqtt/connect Connect
+   * @apiName connect
+   * @apiGroup Zigbee2mqtt
+   */
+  async function connect(req, res) {
+    logger.log('Entering connect step');
+    await zigbee2mqttManager.init();
+    res.json({
+      success: true,
+    });
+  }
+
+  /**
+   * @api {post} /api/v1/service/zigbee2mqtt/mqtt/start Install & start MQTT container.
+   * @apiName installMqttContainer
+   * @apiGroup Zigbee2mqtt
+   */
+  async function installMqttContainer(req, res) {
+    logger.log('Install MQTT container');
+    let response;
+    try {
+      await zigbee2mqttManager.installMqttContainer();
+      response = true;
+    } catch (e) {
+      logger.error('Error while connecting to MQTT:', e);
+      response = false;
+      gladys.event.emit(EVENTS.WEBSOCKET.SEND_ALL, {
```

Comment:
Not really fan of functional stuffs in controller.

Reply:
Removed (it's already there in installMqttContainer)